### PR TITLE
Start implementing viewport interaction item_left

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This is the final part of @marijnvdwerf interaction item_left that needed to be implemented. I needed more knowledge from get map position before being able to tackle it confidently.

### CODE REVIEW

File: src/OpenLoco/Station.cpp [hunk 1]
```diff
@@ -779,4 +776,27 @@ namespace OpenLoco
 
         setStationCatchmentRegion(cargoSearchState, minPos, maxPos, flag);
     }
+
+    string_id getTransportIconsFromStationFlags(const uint16_t flags)
+    {
+        static const string_id label_icons[] = {
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Station.cpp:782]: Any particular reason the `constexpr` was removed?

Comment (duncanspumpkin) [reply src/OpenLoco/Station.cpp:782]: There wasn't much point in it after moving it to the cpp file. Guess it doesn't really matter when in the cpp file.

Comment (duncanspumpkin) [reply src/OpenLoco/Station.cpp:782]: I've reverted it back to a constexpr

**Comment by duncanspumpkin** [comment]
This is now working but I'll refactor it now and move functions to more appropriate files.

**Comment by IntelOrca** [comment]
The functions that start with underscore... I assume you did it because it conflicts with structs, but is your plan to keep them or rename the structs or should you just rename the functions?

**Comment by duncanspumpkin** [comment]
it was how @marijnvdwerf had originally written the functions. I'll go through and rename them.

**Comment by AaronVanGeffen** [comment]
This PR generally looks good to me, but testing revealed a regression. When clicking a track element of an _underground_ station, the wrong station appears to open. Note that clicking the station name works fine; this just concerns the track elements.

In the save file attached, I have centred the viewport around such a station. Instead of opening 'Salt Lake City Sidings', 'Phoenix Central' is opened. 

[North America (West) 1955a.SV5.zip](https://github.com/OpenLoco/OpenLoco/files/5826666/North.America.West.1955a.SV5.zip)

**Comment by duncanspumpkin** [comment]
What do people think of that last commit? Is that going to be worse or not?

File: src/OpenLoco/Ui/ViewportInteraction.cpp [hunk 2]
```diff
@@ -104,11 +104,7 @@ namespace OpenLoco::Ui::ViewportInteraction
         Input::setMapSelectionFlags(Input::MapSelectionFlags::unk_6);
         ViewportManager::invalidate(station);
         Windows::MapToolTip::setOwner(station->owner);
-        auto args = Windows::MapToolTip::getArguments();
-        args.push(StringIds::stringid_stringid_wcolour3_stringid);
-        args.push(station->name);
-        args.push(station->town);
-        args.push(getTransportIconsFromStationFlags(station->flags));
+        auto args = FormatArguments::mapToolTip(StringIds::stringid_stringid_wcolour3_stringid, station->name, station->town, getTransportIconsFromStationFlags(station->flags));
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Ui/ViewportInteraction.cpp:107]: Not a fan of the formatting here. LGTM otherwise.

File: src/OpenLoco/Map/Tile.h [hunk 3]
```diff
@@ -259,7 +262,7 @@ namespace OpenLoco::Map
         uint8_t _slope;
         uint8_t _water;
         uint8_t _terrain;
-        uint8_t _industry;
+        uint8_t getIndustryArguments;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Map/Tile.h:265]: revert rename

File: src/OpenLoco/Station.cpp [hunk 4]
```diff
@@ -34,7 +34,7 @@ namespace OpenLoco
         inline static loco_global<uint32_t, 0x0112C68C> _filter;
         inline static loco_global<uint32_t[max_cargo_stats], 0x0112C690> _score;
         inline static loco_global<uint32_t, 0x0112C710> _producedCargoTypes;
-        inline static loco_global<industry_id_t[max_cargo_stats], 0x0112C7D2> _industry;
+        inline static loco_global<industry_id_t[max_cargo_stats], 0x0112C7D2> getIndustryArguments;
```
Comment (duncanspumpkin) [inline src/OpenLoco/Station.cpp:37]: revert rename

**Comment by duncanspumpkin** [comment]
Okay I think I've fixed all of the issues with this now. I'm not getting the wrong station now or a few other station issues.

**Review by AaronVanGeffen — APPROVED** [review]
I've re-tested this with today's changes. As of 1cbac86cb4f0da1d29d00de4b74a3e19ffbdd8b9, all interaction appears to be working properly. Nice work!

**Pull request merged**